azure/services/aso/aso.go

```diff
@@ -141,6 +148,9 @@ func (s *Service) CreateOrUpdateResource(ctx context.Context, spec azure.ASOReso
 
 	if existing == nil {
 		labels[infrav1.OwnedByClusterLabelKey] = s.clusterName
+		annotations[ReconcilePolicyAnnotation] = ReconcilePolicySkip
```

Comment:
One possible optimization that could be added is not to update ASO resources if `reconcile-policy: skip` is set since those changes aren't meant to propagate to the Azure resource anyway. I wonder if it might be valuable to do updates anyway though since that combined with the `status` would describe "what CAPZ wants the resource to look like" vs. "how it actually looks."

Reply:
So that I follow... in the nominal "new resource" case where we are creating a new ASO resource for the first time, we want to set the `"serviceoperator.azure.com/reconcile-policy"` annotation to `"skip"`? Why is that?

Reply:
Yep. This is to handle the "resource already exists in Azure but not managed by ASO" case, in which we assume the user is managing the resource themselves so ASO shouldn't make any changes to it. This is essentially to facilitate ASO doing its initial GET to Azure so we can see its status.

Reply:
Does this mean we are sort of "initializing" to skip and then will plan to reset in the next reconciliation when we realize that the resource doesn't in fact exist already in Azure?

Reply:
Yep, exactly. That happens here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/45dc9456f266b920225b3682ac4ca0d2a20ed503/azure/services/aso/aso.go#L104-L109

Reply:
O.K. cool. Can we add code comments to describe what we're doing here and why? With these kinds of runloops the "init"/"base case" stuffs are often (IME) non-intuitive, especially when you're reading for the first time and sort of relying upon the literal execution loop and its conditional expressions for understanding "this because that". In fact it's more like "this right now because that later responds to this during the next loop" (sorry, I'm not making things clearer!)

Reply:
Just added a couple comments. LMK if that makes things clearer.